arch/risc-v/src/common/riscv_mtimer.c

```diff
@@ -0,0 +1,119 @@
+/****************************************************************************
+ * arch/risc-v/src/common/riscv_mtimer.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/arch.h>
+
+#include <arch/barriers.h>
+
+#include <stdint.h>
+
+#include "riscv_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define is_lt_u64(a, b)     ((int64_t)((int64_t)(a) - (int64_t)(b)) < 0)
```

Comment:
```suggestion
#define is_lt_u64(a, b)     (((int64_t)(a) - (int64_t)(b)) < 0)
```

Reply:
The result needs casting, the incoming parameters themselves don't. I can remove the parameter casting?

Reply:
Done, please take a look

Reply:
if `int64_t` cast is applied to both `a` and `b`, then the result is also the same type `int64_t` isn't it?

Reply:
> Done, please take a look

This way also should work

Reply:
Yes both solutions would work, the other implicitly does promotion to long long int:

_If the types are the same, that type is the common type._ 

But the parameters do not need casting, the arithmetic can be unsigned as long as the result is casted. This is IMO clearer :)

Reply:
yes. let's keep parameter type arithmetic and then cast a result